src/extension.ts

```diff
@@ -1,27 +1,29 @@
 // The module 'vscode' contains the VS Code extensibility API
 // Import the module and reference it with the alias vscode in your code below
 import * as vscode from 'vscode';
+import { showReferrers } from './commands/showReferrer';
 
-// This method is called when your extension is activated
-// Your extension is activated the very first time the command is executed
-export function activate(context: vscode.ExtensionContext) {
-
-	// Use the console to output diagnostic information (console.log) and errors (console.error)
-	// This line of code will only be executed once when your extension is activated
-	console.log('Congratulations, your extension "annaPrototype" is now active!');
-
-	// The command has been defined in the package.json file
-	// Now provide the implementation of the command with registerCommand
-	// The commandId parameter must match the command field in package.json
-	let disposable = vscode.commands.registerCommand('annaPrototype.helloWorld', () => {
-		// The code you place here will be executed every time your command is executed
-		// Display a message box to the user
-		vscode.window.showInformationMessage('Hello World from Prototype!');
-	});
-
-	context.subscriptions.push(disposable);
+/**  This interface declares that any object of type IremoteTreeItem should have a 
+ * fullTag property of type string. It defines the type of an object that 
+ * is expected to be passed to our command async callback function.
+ */  
+export interface IRemoteTagTreeItem {
+	readonly fullTag: string;
+	readonly parent: any;
 }
 
 
-// This method is called when your extension is deactivated
-export function deactivate() {}
+export function activate(context: vscode.ExtensionContext ) {
+/** Registering the command with id 'securesupplychain.showReferrers' and giving it a async callback function
```

Comment:
This is pretty standard, and the comment is not helpful to understand the immediate function, rather it tries to explain how it is internally implemented. It can be removed.